compiler/resolution/interfaceResolution.cpp

```diff
@@ -391,24 +467,18 @@ static bool resolveOneWitness(InterfaceSymbol* isym, ImplementsStmt* istm,
 
   // Resolve the call and see if we got a witness.
   FnSymbol* target = tryResolveCall(call);
+  call->remove();
+
   if (target != NULL) {
     cgprint("%s           -> %s  %s\n", indent,
             symstring(target), debugLoc(target));
     resolveFunction(target); // aborts if there are errors
 
-    call->remove();
-    if (holder->body.tail != NULL) {
-      if (reportErrors) {
-        USR_FATAL_CONT(istm, "when checking this implements statement");
-        USR_PRINT(target, "this implementation of the required function %s"
-          " requires implicit conversion(s), which is currently disallowed",
-          reqFn->name);
-        USR_PRINT(reqFn, "the required function %s in the interface %s"
-                         " is declared here", reqFn->name, isym->name);
-      }
+    if (!( checkNonemptyHolder(isym, istm, holder, target, reqFn, reportErrors)
+           & checkReturnType(isym, istm, target, fml2act, reqFn, reportErrors)
+           & checkReturnIntent(isym, istm, target, reqFn, reportErrors) ))
```

Comment:
It seems odd that you are using `&` and not `&&` here - did you really mean that?